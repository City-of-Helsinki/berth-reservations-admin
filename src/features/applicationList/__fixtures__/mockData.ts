import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
