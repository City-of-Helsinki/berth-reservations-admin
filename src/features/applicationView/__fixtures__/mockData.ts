import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
