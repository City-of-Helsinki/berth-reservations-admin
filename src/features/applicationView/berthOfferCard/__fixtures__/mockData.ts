import { BerthLease_lease as BerthLease } from '../__generated__/BerthLease';
import {
  AdditionalProductType,
  BerthMooringType,
  LeaseStatus,
  ProductServiceType,
} from '../../../../@types/__generated__/globalTypes';

export const lease: BerthLease = {
  id: '165023f5-5b74-4656-9315-d4e1a22f111d',
  status: LeaseStatus.DRAFTED,
  berth: {
    depth: null,
    length: 6,
    mooringType: BerthMooringType.SINGLE_SLIP_PLACE,
    width: 2.5,
    comment: '',
    isAccessible: null,
    number: '1',
    pier: {
      properties: {
        identifier: 'C',
        electricity: true,
        gate: true,
        lighting: true,
        mooring: true,
        wasteCollection: true,
        water: true,
        harbor: {
          id: 'b55da996-cf00-4b52-9aa0-6b13fd8d806b',
          properties: {
            name: 'Aurinkolahden venesatama (Aurinkoranta)',
            __typename: 'HarborProperties',
          },
          __typename: 'HarborNode',
        },
        __typename: 'PierProperties',
      },
      __typename: 'PierNode',
    },
    __typename: 'BerthNode',
  },
  customer: {
    __typename: 'ProfileNode',
    id: 'MOCK-CUSTOMER',
    primaryEmail: {
      __typename: 'EmailNode',
      id: 'MOCK-EMAIL',
      email: 'test@example.com',
    },
  },
  order: {
    id: '07f65f54-4a25-47fd-9eea-d9fdddfa46f9',
    orderNumber: '123456',
    price: '96.67',
    totalPrice: '256.70',
    orderLines: {
      edges: [
        {
          node: {
            id: '9b97b9bc-bba5-46d9-aae2-86e8686012ef',
            price: '20.50',
            product: {
              id: 'dc7f8158-a4cb-49a9-9368-e586f224fdb5',
              service: ProductServiceType.ELECTRICITY,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ea2ef175-f790-45a0-b16f-9e5a98340691',
            price: '19.33',
            product: {
              id: '2e55f605-061e-4299-814d-bf2cd53aa8ce',
              service: ProductServiceType.WATER,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '8ba4de61-63cc-49fc-afe2-26068f3c8e4d',
            price: '2.42',
            product: {
              id: '3a2e3a2f-2782-40e9-a139-b9a4d4a36c3b',
              service: ProductServiceType.GATE,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '59c768c1-f371-4178-afda-1e16d2c725b0',
            price: '8.33',
            product: {
              id: '762da6e9-89be-43bc-b097-53dff4ed8415',
              service: ProductServiceType.MOORING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '6d910d8e-80b6-4486-aa85-4958bcd98dd5',
            price: '4.17',
            product: {
              id: '34c5b6bc-7eb8-457e-bce8-8ba3a07b9057',
              service: ProductServiceType.WASTE_COLLECTION,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ab7cf1e5-92b5-42ad-844a-e2d577d30baf',
            price: '40.28',
            product: {
              id: '650dbfcb-8f10-46b6-af64-197b77686a5b',
              service: ProductServiceType.LIGHTING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '244a7af9-6f91-4f08-9738-82ef06a2e790',
            price: '65.00',
            product: {
              id: '5194dd14-0226-4603-808c-831cfbec72de',
              service: ProductServiceType.DINGHY_PLACE,
              productType: AdditionalProductType.OPTIONAL_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
      ],
      __typename: 'OrderLineNodeConnection',
    },
    __typename: 'OrderNode',
  },
  __typename: 'BerthLeaseNode',
};
