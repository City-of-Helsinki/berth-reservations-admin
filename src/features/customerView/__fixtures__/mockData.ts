import { OrderOrderType, OrderStatus, PriceUnits, ProductServiceType } from '../../../@types/__generated__/globalTypes';
import { BerthInvoice, WinterStorageInvoice } from '../types';
import { INDIVIDUAL_CUSTOMER_profile as CUSTOMER_PROFILE } from '../__generated__/INDIVIDUAL_CUSTOMER';

export const emptyMockProfile: CUSTOMER_PROFILE = {
  __typename: 'ProfileNode',
  berthApplications: null,
  berthLeases: null,
  winterStorageLeases: null,
  boats: null,
  comment: null,
  firstName: '',
  id: '',
  invoicingType: null,
  language: null,
  lastName: '',
  orders: null,
  organization: null,
  primaryAddress: null,
  primaryEmail: null,
  primaryPhone: null,
  customerGroup: null,
};

export const mockInvoices: (BerthInvoice | WinterStorageInvoice)[] = [
  {
    orderNumber: 'MOCK-INVOICE',
    orderType: OrderOrderType.LEASE_ORDER,
    status: OrderStatus.WAITING,
    berthInformation: {
      number: '1',
      pierIdentifier: 'C',
      harborName: 'Aurinkolahden venesatama (Aurinkoranta)',
    },
    contractPeriod: {
      startDate: '2020-06-26',
      endDate: '2020-09-14',
    },
    dueDate: '2020-07-10',
    basePrice: 96.67,
    totalPrice: 191.7,
    orderLines: [
      {
        product: ProductServiceType.ELECTRICITY,
        price: 20.5,
        priceUnit: PriceUnits.PERCENTAGE,
        priceValue: 9,
      },
    ],
  },
];
