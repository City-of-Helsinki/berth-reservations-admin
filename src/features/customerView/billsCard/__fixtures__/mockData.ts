import { ProductServiceType } from '../../../../@types/__generated__/globalTypes';

export const mockBills = [
  {
    berthInformation: {
      number: 1,
      pierIdentifier: 'C',
      harborName: 'Aurinkolahden venesatama (Aurinkoranta)',
    },
    contractPeriod: {
      startDate: '2020-06-26',
      endDate: '2020-09-14',
    },
    dueDate: '2020-07-10',
    basePrice: 96.67,
    basePriceTaxPercentage: 24.0,
    totalPrice: 191.7,
    totalPriceTaxPercentage: 19.36,
    orderLines: [
      {
        product: ProductServiceType.ELECTRICITY,
        price: 20.5,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.WATER,
        price: 19.33,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.GATE,
        price: 2.42,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.MOORING,
        price: 8.33,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.WASTE_COLLECTION,
        price: 4.17,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.LIGHTING,
        price: 40.28,
        taxPercentage: 24.0,
      },
    ],
  },
];
