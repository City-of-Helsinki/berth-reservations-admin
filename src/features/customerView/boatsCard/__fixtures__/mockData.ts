import { Boat, BoatCertificate, LargeBoat } from '../../types';
import { BoatCertificateType } from '../../../../@types/__generated__/globalTypes';

export const insuraceCert: BoatCertificate = {
  file: 'https://venepaikka-api.test.hel.ninja/media/boats/2c736de2-5659-46ea-ad3f-9c1364307d7d/dummy_pdf2.pdf',
  certificateType: BoatCertificateType.INSURANCE,
  validUntil: null,
  checkedAt: '2020-04-28',
  checkedBy: null,
};

export const inspectionCert: BoatCertificate = {
  file: null,
  certificateType: BoatCertificateType.INSPECTION,
  validUntil: '2099-09-04',
  checkedAt: '2020-04-28',
  checkedBy: 'Jukka Virtanen',
};

export const boat: Boat = {
  id: '1111',
  boatType: {
    id: '7',
    name: 'Troolari',
  },
  width: 2,
  length: 10,
  draught: 2,
  weight: 15000,
  name: 'Huh hah hei',
  model: 'Trawl 11',
  registrationNumber: 'TRLR-555',
};

export const largeBoat: LargeBoat = {
  id: '9999',
  boatType: {
    id: '8',
    name: 'Suuri alus (yli 20t)',
  },
  width: 4,
  length: 15,
  draught: 2,
  weight: 15000,
  name: 'DevilsDriver',
  model: 'Trawl 77',
  registrationNumber: 'TRLR-666',
  propulsion: 'Bensin',
  hullMaterial: 'Aluminium',
  intendedUse: 'For fun',
  certificates: [inspectionCert, insuraceCert],
};
