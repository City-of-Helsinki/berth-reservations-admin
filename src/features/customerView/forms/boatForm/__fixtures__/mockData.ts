import { BoatType } from '../../../../harborView/forms/types';
import { Boat } from '../../../types';

export const mockBoatTypes: BoatType[] = [{ id: '1', name: 'Purjevene' }];

export const mockValidValues: Boat = {
  boatType: mockBoatTypes[0],
  id: 'id',
  model: 'model',
  name: 'name',
  registrationNumber: '123456',
  weight: 1,
  width: 1,
  length: 1,
  draught: 1,
};

export const mockInvalidValues: Boat = {
  boatType: mockBoatTypes[0],
  id: 'id',
  model: 'model',
  name: 'name',
  registrationNumber: '',
  weight: -10,
  width: 1,
  length: 1,
  draught: 1,
};
