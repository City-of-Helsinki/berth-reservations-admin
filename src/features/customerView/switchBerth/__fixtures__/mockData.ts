import { HarborOption } from '../types';
import { SWITCH_BERTH_HARBORS } from '../__generated__/SWITCH_BERTH_HARBORS';

export const mockOption1: HarborOption = {
  label: 'Aataminsatama',
  value: '3',
};

export const mockOption2: HarborOption = {
  label: 'Albertinsatama',
  value: '2',
};

export const mockOption3: HarborOption = {
  label: 'Bertansatama',
  value: '4',
};

export const mockOption4: HarborOption = {
  label: 'Zorronsatama',
  value: '1',
};

export const mockSwitchBerthHarbors: SWITCH_BERTH_HARBORS = {
  harbors: {
    __typename: 'HarborNodeConnection',
    edges: [
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '1',
          properties: { __typename: 'HarborProperties', name: 'Zorronsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '2',
          properties: { __typename: 'HarborProperties', name: 'Albertinsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '3',
          properties: { __typename: 'HarborProperties', name: 'Aataminsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '4',
          properties: { __typename: 'HarborProperties', name: 'Bertansatama' },
        },
      },
    ],
  },
};
