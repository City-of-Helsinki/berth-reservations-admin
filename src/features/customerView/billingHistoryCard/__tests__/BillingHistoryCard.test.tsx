import React from 'react';
import { shallow } from 'enzyme';

import BillingHistoryCard from '../BillingHistoryCard';
import { mockBills } from '../../__fixtures__/mockData';

describe('BillingHistoryCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<BillingHistoryCard bills={mockBills} onClick={jest.fn} />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
