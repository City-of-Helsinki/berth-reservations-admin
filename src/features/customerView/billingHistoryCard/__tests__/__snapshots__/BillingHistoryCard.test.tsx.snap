// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingHistoryCard renders correctly 1`] = `
<div
  class="card"
>
  <div
    class="cardHeader"
  >
    <span
      class="title"
    >
      LASKUHISTORIA
    </span>
  </div>
  <div
    class="cardBody"
  >
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        LASKUT
      </h4>
      <section
        class="body"
      >
        <div
          class="grid cols4"
        >
          <button
            class="gridItem"
          >
            <span
              class="text brand span"
            >
              Venepaikan vuokra
            </span>
          </button>
          <div
            class="gridItem"
          >
            <span
              class="text standard span"
            >
              10.07.2020
            </span>
          </div>
          <div
            class="gridItem"
          >
            <span
              class="text standard span"
            >
              191,70 €
            </span>
          </div>
          <span
            class="chip orange"
          >
            Odottaa maksua
          </span>
        </div>
      </section>
    </article>
  </div>
   
</div>
`;
